### ISSUE TITLE
Writing 0 twice to the first address on inner loop

### ISSUE THREAD

**Issue by mahircg** [ilirlikalla/EMPLab_15#1]
During last week's demonstration,as we all seen, 0 is written twice to the first address of the inner loop of ;

 for( i=2; i<=N/2; i++ )                                       
   for( j=i+i; j<=N; j+=i )                                    
     a[j] = 0;

At first,or until now, I though it is a bug related to the decoding logic.However,after looking at the assembly program closely, I have seen that it is an issue with the code itself.

inner:    sb    0(r7), r0     ; A[j] = 0                 
            sle   r8, r6, r2    ; r8: j <= N ?              
            addu  r7, r3, r6    ; r7 : Adresse von A[j]       
            addu  r6, r6, r4    ; j = j + i                   
            bnez  r8, inner     ; if r8 == 1 loop inner     

sle instruction on second line checks if J is less than N. And decision of branching or not is done upon output of that instruction, meaning that the same address will be used again in next iteration, and the incremented index will be checked agains N on the very same(second) iteration.

Long story short,strange(or supposed) behavior is caused by the code. Maybe we should let the lab assistants know about it (or do they already know o_O)
